pkg/controller/v1alpha2/experiment/experiment_controller.go

```diff
@@ -58,7 +62,21 @@ func Add(mgr manager.Manager) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileExperiment{Client: mgr.GetClient(), scheme: mgr.GetScheme()}
+	r := &ReconcileExperiment{
+		Client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+	}
+	// If the flag is set in CLI, use the fake implementation.
+	imp := viper.GetString(consts.ConfigFakeExperimentSuggestion)
+	switch imp {
+	case "fake":
+		log.Info("Using the fake suggestion implementation")
+		r.Suggestion = suggestionfake.New()
+	default:
```

Comment:
if user sets it as "invalid-one", we can still use default, but a warning log is necessary

Reply:
SGTM